service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapper.java

```diff
@@ -179,14 +181,15 @@ private String buildDoseQuantityPertinentInformation(Immunization immunization)
     }
 
     private String buildNotePertinentInformation(Immunization immunization) {
-        String notes = StringUtils.EMPTY;
-        if (immunization.hasNote()) {
-            List<Annotation> annotations = immunization.getNote();
-            notes = annotations.stream()
-                .map(Annotation::getText)
-                .collect(Collectors.joining(StringUtils.SPACE));
-        }
-        return notes;
+        return Stream.concat(
+            immunization.hasNote() ? immunization.getNote().stream() : Stream.empty(),
+            messageContext.getInputBundleHolder().getRelatedConditions(immunization.getId())
+                .stream()
+                .map(Condition::getNote)
+                .flatMap(List::stream)
+        )
```

Comment:
Here is where we need to reverse the order of the to ensure the condition notes are preprended before the immunization note.

```suggestion
            messageContext.getInputBundleHolder().getRelatedConditions(immunization.getId())
                .stream()
                .map(Condition::getNote)
                .flatMap(List::stream)
            ),
            immunization.hasNote() ? immunization.getNote().stream() : Stream.empty()
            
```